tests/TestCase/Authenticator/PasetoAuthenticatorTest.php

```diff
@@ -0,0 +1,553 @@
+     */
+    public function testAuthenticationPayloadIsNull(): void
+    {
+        $request = ServerRequestFactory::fromGlobals(
+            ['REQUEST_URI' => '/']
+        );
+        $token = $this->buildLocalToken(new Version4());
+        $request = $request->withAddedHeader('Authorization', 'Bearer ' . $token->toString());
+
+        $authenticator = $this->getMockBuilder(PasetoAuthenticator::class)
+            ->setConstructorArgs([
+                $this->identifiers,
+                [
+                    'purpose' => PasetoAuthenticator::LOCAL,
+                    'version' => 'v4',
+                ],
+            ])
+            ->onlyMethods([
+                'getPayLoad',
+            ])
+            ->getMock();
+
+        $authenticator->expects($this->once())
+            ->method('getPayLoad')
+            ->willReturn(null);
+
+        $result = $authenticator->authenticate($request);
+        $this->assertInstanceOf(Result::class, $result);
+        $this->assertSame(Result::FAILURE_CREDENTIALS_INVALID, $result->getStatus());
+        $this->assertNull($result->getData());
+    }
+
+    /**
+     * Test getPayLoad returns null
+     *
+     * @return void
+     * @throws \ParagonIE\Paseto\Exception\InvalidKeyException
+     * @throws \ParagonIE\Paseto\Exception\InvalidPurposeException
+     * @throws \ParagonIE\Paseto\Exception\PasetoException
+     */
+    public function testAuthenticationReturnsNullPayload(): void
+    {
+        $authenticator = new PasetoAuthenticator($this->identifiers, [
+            'secretKey' => self::LOCAL_SECRET_KEY,
+            'purpose' => PasetoAuthenticator::LOCAL,
+            'version' => 'v4',
+        ]);
+
+        $this->assertNull($authenticator->getPayload());
+    }
+
+    /**
+     * Test getPayLoad returns a JsonObject with no subject.
+     *
+     * @return void
+     * @throws \ParagonIE\Paseto\Exception\InvalidKeyException
+     * @throws \ParagonIE\Paseto\Exception\InvalidPurposeException
+     * @throws \ParagonIE\Paseto\Exception\PasetoException
+     */
+    public function testAuthenticationReturnsJsonObjectWithoutSub(): void
+    {
+        $request = ServerRequestFactory::fromGlobals(
+            ['REQUEST_URI' => '/']
+        );
+        $token = $this->buildLocalToken(new Version4());
+        $request = $request->withAddedHeader('Authorization', 'Bearer ' . $token->toString());
+
+        $authenticator = $this->getMockBuilder(PasetoAuthenticator::class)
+            ->setConstructorArgs([
+                $this->identifiers,
+                [
+                    'purpose' => PasetoAuthenticator::LOCAL,
+                    'version' => 'v4',
+                ],
+            ])
+            ->onlyMethods([
+                'getPayLoad',
+            ])
+            ->getMock();
+
+        $authenticator->expects($this->once())
+            ->method('getPayLoad')
+            ->willReturn((new JsonToken())->setSubject(''));
```

Comment:
Does this need to use a mock? There is a lot of code being dedicated to mocking to avoid creating a value we could generate with fewer lines of code.

Could we try having a helper method that generates a key pair and creates an authenticator using that key pair. You would also need a method that generates a token for a given input.

This is a bit of plumbing work to do but in the long term it helps with maintenance as we don't have to worry about changes in a mocking library. Additionally it also gives us more confidence in the tests as we are testing more of the subsytem all together.